Sources/ArcGISToolkit/Components/Form/FormInputs/SingleLineTextInput.swift

```diff
@@ -77,10 +89,77 @@ struct SingleLineTextInput: View {
             }
         }
         .onChange(of: text) { newValue in
-            guard newValue != element.value else {
-                return
+            // Note: this will be replaced by `element.updateValue()`, which will
+            // handle all the following logic internally.
+            if isDecimal {
+                // Note: this should handle other decimal types as well, if they exist (float?)
+                let value = Double(newValue)
+                featureForm?.feature.setAttributeValue(value, forKey: element.fieldName)
+            } else if isNumeric {
+                // Note: this should handle more than just Int32
+                let value = Int32(newValue)
+                featureForm?.feature.setAttributeValue(value, forKey: element.fieldName)
+            } else {
+                // Text field
+                featureForm?.feature.setAttributeValue(newValue, forKey: element.fieldName)
             }
-            featureForm?.feature.setAttributeValue(newValue, forKey: element.fieldName)
         }
     }
 }
+
+private extension SingleLineTextInput {
+    /// A Boolean value indicating whether the input is for a numeric data type.
+    var isNumeric: Bool {
+        if let field = featureForm?.feature.table?.field(named: element.fieldName)?.type {
+            return field.isNumeric
+        }
+        return false
+    }
+    
+    /// A Boolean value indicating whether the field has a numeric data type with decimal precision.
+    var isDecimal: Bool {
+        if let field = featureForm?.feature.table?.field(named: element.fieldName)?.type {
+            return field.isFloatingPoint
+        }
+        return false
+    }
+    
+    /// The keyboard type to use depending on where the input is numeric and decimal.
+    var keyboardType: UIKeyboardType {
+        isNumeric ? (isDecimal ? .decimalPad : .numberPad) : .default
+    }
+    
+    /// The button that allows a user to switch the numeric value between positive and negative.
+    var positiveNegativeButton: some View {
+        Button {
+            if text.first == "-" {
+                text.removeFirst()
+            } else {
+                text.insert("-", at: text.startIndex)
+            }
+        } label: {
+            Image(systemName: "plus.forwardslash.minus")
+        }
+    }
+    
+    /// The range of valid values for a numeric input field.
+    var rangeDomain: RangeDomain? {
+        if let field = featureForm?.feature.table?.field(named: element.fieldName) {
+            return field.domain as? RangeDomain
+        } else {
+            return nil
+        }
+    }
+}
+
+private extension FieldType {
```

Comment:
If we end up using this in more than one file, move it to it's own file in the Extension folder.